test/test.js

```
@@ -115,10 +116,23 @@ describe('node-couchdb tests', () => {
             });
     });
 
+    it('should reject with EBADREQUEST if database name contains special character', () => {
+        dbName += '$§%&/'
+        return couch.createDatabase(dbName)
+            .then(() => {
+                throw new Error(`Creating of the database ${dbName} with special character didn't fail`);
+            }, err => {
+                assert.instanceOf(err, Error, 'rejected error is not an instance of Error');
+                assert.strictEqual(err.code, 'EBADREQUEST');
+            });
+    });
+
     // dropDatabase() operations
     it('should return promise for dropDatabase operation', () => {
-        const promise = couch.dropDatabase(dbName);
+        const promise = couch.dropDatabase(dbName)
+            .catch(function(){ /* no error handling */ });
```

The newest node version throws a warning when a promise was created and not handled:
> UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Database n
ot found: sample1485607844453

So I just returned the Promise to mocha.

The dropDatabase throws an error because the database which we want to delete is not existing.
So the test fails.
But because we are not interested in this error only in the return type i implemented an empty catch block.